centreon/www/front_src/src/AppTokens/api/decoder.ts

```diff
@@ -0,0 +1,34 @@
+import { JsonDecoder } from 'ts.data.json';
+
+import { buildListingDecoder } from '@centreon/ui';
+
+import { PersonalInformation, Token } from '../TokenListing/models';
+
+const personalInformationDecoder = (
+  decoderName = 'personalInformation'
+): JsonDecoder.Decoder<PersonalInformation> =>
+  JsonDecoder.object<PersonalInformation>(
+    {
+      id: JsonDecoder.number,
+      name: JsonDecoder.string
+    },
+    decoderName
+  );
+
+const tokenDecoder = JsonDecoder.object<Token>(
+  {
+    creation_date: JsonDecoder.string,
+    creator: personalInformationDecoder('creator'),
+    expiration_date: JsonDecoder.string,
+    is_revoked: JsonDecoder.boolean,
+    name: JsonDecoder.string,
+    user: personalInformationDecoder('user')
+  },
+  'Token'
+);
```

Comment:
you can decode expiration_date and is_revoked to expirationDate and isRevoked as well